src/lxc/tools/lxc_unshare.c

```diff
@@ -112,6 +117,9 @@ static int do_start(void *arg)
 	int want_default_mounts = start_arg->want_default_mounts;
 	const char *want_hostname = start_arg->want_hostname;
 
+	/* waiting until uid maps is set */
+	read(wait_fd, &val, sizeof(char));
```

Comment:
You can't really ignore the return value from `read()`/`write()`. You need to assign it and do something with it you could `printf()` a harmless info or direct output of `fprintf()` to `/dev/null`:

```
	mv -f $depbase.Tpo $depbase.Po
../../../src/lxc/tools/lxc_unshare.c: In function ‘main’:
../../../src/lxc/tools/lxc_unshare.c:276:8: error: ignoring return value of ‘write’, declared with attribute warn_unused_result [-Werror=unused-result]
   write(wait_fd, &val, 8);
        ^
../../../src/lxc/tools/lxc_unshare.c: In function ‘do_start’:
../../../src/lxc/tools/lxc_unshare.c:121:6: error: ignoring return value of ‘read’, declared with attribute warn_unused_result [-Werror=unused-result]
  read(wait_fd, &val, sizeof(char));
```